### PR TITLE
Fix nil check in map merge

### PR DESCRIPTION
**Pull request by astraw99**
#### What type of PR is this?
/kind bug

#### What this PR does / why we need it:
Fix nil check in map merge, or nil map will lead to panic.

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

### CODE REVIEW

**Comment by astraw99** [comment]
/assign @davidz627
PTAL thanks.

**Review by Jiawei0227 — APPROVED** [review]
/lgtm

File: pkg/volume/csi/csi_mounter.go [hunk 1]
```diff
@@ -477,8 +477,12 @@ func mergeMap(first, second map[string]string) map[string]string {
 	if first == nil && second != nil {
 		return second
 	}
-	for k, v := range second {
-		first[k] = v
```
[the remark this answers is not in the document]
Comment (jsafrane) [reply pkg/volume/csi/csi_mounter.go:481]: I am sorry, but no. The code looks worse with nested `for` in `if`.

What about fixing it better? The first condition can be:
```
if first == nil {
    return second
}
```

`second` is either non-nil and that's returned today (good) or `nil`, which is the same `nil` as would be returned by `return first` later, so looks good too.

And keep the `for` loop as it is.

[the remark this answers is not in the document]
Comment (astraw99) [reply pkg/volume/csi/csi_mounter.go]: Updated, looks better now :) 
PTAL, thanks.

**Pull request merged**